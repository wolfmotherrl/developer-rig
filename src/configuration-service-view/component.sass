.configuration-service-view
  display: grid
  grid-template-columns: 1fr 8pt 1fr
  grid-gap: 8pt
  margin-top: 8pt

  &__button
    background-color: #6441A4
    border: 0
    color: white
    height: 15pt
    margin-left: 8pt
    min-width: 8em

    &--first
      margin-left: 0

  &__title
    font-size: larger
    font-weight: bold

  &-property
    display: block
    margin: 1em

    &__input
      width: 70%

      &--error
        border-color: red

    &__name
      margin-bottom: 5px
      margin-right: 11px

    &__value
      color: grey
      user-select: text

  &__vertical-bar
    background-color: #DAD8DE
    width: 1pt;
    height: 100%

  &__item
    margin-bottom: 8pt
    margin-left: -17pt

  &__item-text
    margin: 3pt

    &--title
      font-weight: bold
