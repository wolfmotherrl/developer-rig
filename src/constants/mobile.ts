export const MobileOrientation: { [key: string]: string } = {
  Portrait: 'Portrait',
  Landscape: 'Landscape',
}

export const DefaultMobileOrientation = MobileOrientation.Portrait;

export const MobileSizes: { [key: string]: { width: number; height: number; } } = Object.freeze({
  'iPhone X/XS Max (414x896)': {
    width: 414,
    height: 896
  },
  'iPhone X/XS (375x822)': {
    width: 375,
    height: 822
  },
  'iPhone 7/8 (375x667)': {
    width: 375,
    height: 667
  },
  'iPhone 7/8 Plus (414x736)': {
    width: 414,
    height: 736
  },
  'Google Pixel 2 (411x731)': {
    width: 411,
    height: 731
  },
  'Samsung Galaxy S8/S9 (360x740)': {
    width: 360,
    height: 740
  },
  'One Plus 3 (480x853)': {
    width: 480,
    height: 853
  },
});

export const DefaultMobileSize = 'iPhone X/XS (375x822)';

