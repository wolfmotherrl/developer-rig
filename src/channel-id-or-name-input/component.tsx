import * as React from 'react';
import { DeveloperRigUserId } from '../constants/rig';

interface Props {
  isUser?: boolean;
  name: string;
  value: string;
  labelClassName?: string;
  textClassName?: string;
  onChange: (event: React.FormEvent<HTMLInputElement>) => void;
}

interface State {
  checked: boolean;
  lastChannelId: string;
}


export class ChannelIdOrNameInput extends React.Component<Props, State> {
  state: State = {
    checked: this.props.value === DeveloperRigUserId,
    lastChannelId: '',
  }

  public render() {
    return (
      <>
        <label className={this.props.labelClassName}>
          <span className={this.props.textClassName}>
            {this.props.isUser ? 'User' : 'Channel'} ID or Name
          </span>
          <input type="text" name={this.props.name} value={this.props.value} onChange={this.props.onChange} disabled={this.state.checked} />
        </label>
        <label className={this.props.labelClassName}>
          <input type="checkbox" name="wantsDeveloperRigUserId" onChange={this.onChange} checked={this.state.checked} />
          Use Developer Rig user
        </label>
      </>
    );
  }

  private onChange = (event: React.FormEvent<HTMLInputElement>) => {
    const { checked } = event.currentTarget;
    if (checked) {
      this.setState({ lastChannelId: this.props.value });
      this.props.onChange({ currentTarget: { name: this.props.name, value: DeveloperRigUserId } } as React.FormEvent<HTMLInputElement>);
    } else {
      this.props.onChange({ currentTarget: { name: this.props.name, value: this.state.lastChannelId } } as React.FormEvent<HTMLInputElement>);
    }
    this.setState({ checked });
  }
}
