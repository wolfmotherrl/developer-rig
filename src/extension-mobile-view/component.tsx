import * as React from 'react';
import { ExtensionFrame } from '../extension-frame';
import { MobileOrientation } from '../constants/mobile';
import { FrameSize } from '../core/models/rig';
import { ExtensionMode, ExtensionViewType } from '../constants/extension-coordinator';

const ViewBackgroundColor = '#322F37';
const AbsolutePosition = 'absolute';

interface ExtensionMobileViewProps {
  extension: ExtensionCoordinator.ExtensionObject;
  installationAbilities: ExtensionCoordinator.ExtensionInstallationAbilities;
  frameSize: FrameSize;
  id: string;
  channelId: string;
  configuration: ExtensionCoordinator.Configuration;
  orientation: string;
  position: ExtensionCoordinator.Position
  role: string;
  isLocal: boolean;
  bindIframeToParent: (iframe: HTMLIFrameElement) => void;
}
type Props = ExtensionMobileViewProps & React.HTMLAttributes<HTMLDivElement>;

export class ExtensionMobileView extends React.Component<Props> {
  public computeFrameStyles(): React.CSSProperties {
    let frameStyles: React.CSSProperties;

    if (this.props.orientation === MobileOrientation.Portrait) {
      const height = Math.floor(this.props.frameSize.height * 0.65);
      frameStyles = {
        width: `${this.props.frameSize.width}px`,
        height: `${height}px`,
        bottom: '0'
      }
    } else {
      const width = Math.floor(this.props.frameSize.height * 0.28);
      frameStyles = {
        width: `${width}px`,
        height: `${this.props.frameSize.width}px`,
        right: '0'
      }
    }

    frameStyles.position = AbsolutePosition;
    return frameStyles;
  }

  public computeViewStyles() {
    let viewStyles: React.CSSProperties;
    if (this.props.orientation === MobileOrientation.Portrait) {
      viewStyles = {
        width: this.props.frameSize.width + 'px',
        height: this.props.frameSize.height + 'px',
      }
    } else {
      viewStyles = {
        width: this.props.frameSize.height + 'px',
        height: this.props.frameSize.width + 'px',
      }
    }

    viewStyles.background = ViewBackgroundColor;
    return viewStyles;
  }
  public render() {
    return (
      <div
        className="view component-view"
        style={this.computeViewStyles()}>
        <div style={this.computeFrameStyles()}>
          <ExtensionFrame
            bindIframeToParent={this.props.bindIframeToParent}
            installationAbilities={this.props.installationAbilities}
            className="view"
            channelId={this.props.channelId}
            configuration={this.props.configuration}
            extension={this.props.extension}
            frameId={`frameid-${this.props.id}`}
            mode={ExtensionMode.Viewer}
            type={ExtensionViewType.Mobile}
            isLocal={this.props.isLocal}
            isPopout={false}
          />
        </div>
      </div>
    );
  }
}
