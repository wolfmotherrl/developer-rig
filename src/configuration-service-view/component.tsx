import * as React from 'react';
import './component.sass';
import { Configurations, RigProject } from '../core/models/rig';
import classNames = require('classnames');
import { fetchChannelConfigurationSegments } from '../util/api';
import { ChannelIdOrNameInput } from '../channel-id-or-name-input';
import { DeveloperRigUserId } from '../constants/rig';
import { fetchIdForUser } from '../util/id';

export interface Props {
  configurations: Configurations;
  rigProject: RigProject,
  userId: string;
  authToken: string;
  saveHandler: (segment: string, channelId: string, content: string, version: string) => void,
}

enum ConfigurationType {
  Broadcaster = 'broadcaster',
  Developer = 'developer',
  Global = 'global',
}

interface State {
  version: string;
  configurationType: ConfigurationType;
  channelId: string;
  configuration: string;
  fetchStatus: string;
  lastConfiguration: string;
  lastVersion: string;
  [key: string]: ConfigurationType | string;
}

export class ConfigurationServiceView extends React.Component<Props, State>{
  public state: State = {
    version: this.props.configurations.globalSegment && this.props.configurations.globalSegment.version || '',
    configurationType: ConfigurationType.Global,
    channelId: DeveloperRigUserId,
    configuration: this.props.configurations.globalSegment && this.props.configurations.globalSegment.content || '',
    fetchStatus: '',
    lastConfiguration: '',
    lastVersion: '',
  };

  private onChange = (input: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>) => {
    const { name, value } = input.currentTarget;
    this.setState({ [name]: value });
  }

  private onChangeConfigurationType = (input: React.FormEvent<HTMLSelectElement>) => {
    const { name, value } = input.currentTarget;
    this.setState({ [name]: value, lastConfiguration: '', lastVersion: '' });
    if (value === ConfigurationType.Global) {
      const { content = '', version = '' } = this.props.configurations.globalSegment || {};
      this.setState({ configuration: content, version });
    } else {
      this.setState({ configuration: '', version: '' });
    }
  }

  private fetchChannelConfiguration = async () => {
    let channelId = this.state.channelId.trim();
    if (channelId) {
      this.setState({ fetchStatus: 'fetching...' });
      try {
        channelId = await fetchIdForUser(this.props.authToken, channelId);
        let segmentMap = this.props.configurations.channelSegments[channelId];
        if (!segmentMap) {
          const { rigProject: { manifest: { id: clientId }, secret }, userId } = this.props;
          segmentMap = await fetchChannelConfigurationSegments(clientId, userId, channelId, secret);
        }
        const segment = this.state.configurationType === ConfigurationType.Broadcaster ?
          segmentMap.broadcaster : segmentMap.developer;
        const { content, version } = segment || { content: '', version: '' };
        this.setState({ configuration: content, lastConfiguration: content, version, lastVersion: version });
        this.setState({ fetchStatus: '' });
      } catch (ex) {
        this.setState({ fetchStatus: ex.message });
      }
    }
  }

  private canSave = (): boolean => {
    const { channelId, configurationType } = this.state;
    if (configurationType !== ConfigurationType.Global && !channelId.trim()) {
      // The non-global configuration segments need a channel ID.
      return false;
    }
    return true;
  }

  private save = async () => {
    if (this.canSave()) {
      const { configuration, configurationType, version } = this.state;
      let { channelId } = this.state;
      try {
        if (configurationType !== ConfigurationType.Global) {
          channelId = await fetchIdForUser(this.props.authToken, channelId);
        }
        this.props.saveHandler(configurationType, channelId, configuration.trim(), version.trim());
      } catch (ex) {
        this.setState({ fetchStatus: ex.message });
      }
    }
  }

  private cancel = () => {
    const { configurations } = this.props;
    const { content: configuration, version } = this.state.configurationType === ConfigurationType.Global ? configurations.globalSegment :
      { content: this.state.lastConfiguration, version: this.state.lastVersion };
    this.setState({ configuration, version });
  }

  private viewDocumentation() {
    window.open('https://dev.twitch.tv/docs/extensions/building/#configuration-service', 'developer-rig-help');
  }

  private viewTutorial() {
    window.open('https://www.twitch.tv/videos/320483709', 'developer-rig-help');
  }

  public render() {
    const versionClassName = classNames('configuration-service-view-property__input', {
      'configuration-service-view-property__input--error': !this.state.version.trim(),
    });
    const configurationClassName = classNames('configuration-service-view-property__input', {
      'configuration-service-view-property__input--error': !this.state.configuration.trim(),
    });
    const extensionUrl = `https://dev.twitch.tv/projects/${this.props.rigProject.manifest.id}/files`;
    return (
      <div className="configuration-service-view">
        <div className="configuration-service-view__section configuration-service-view__section--left">
          <label className="configuration-service-view-property">
            <div className="configuration-service-view-property__name">Configuration Type</div>
            <select className="configuration-service-view-property__select" name="configurationType" value={this.state.configurationType} onChange={this.onChangeConfigurationType}>
              <option value={ConfigurationType.Broadcaster}>Broadcaster</option>
              <option value={ConfigurationType.Developer}>Developer</option>
              <option value={ConfigurationType.Global}>Global</option>
            </select>
          </label>
          {this.state.configurationType !== ConfigurationType.Global && <>
            <ChannelIdOrNameInput name="channelId" value={this.state.channelId} onChange={this.onChange}
              labelClassName="configuration-service-view-property" textClassName="configuration-service-view-property__name" />
            <button className="configuration-service-view__button" onClick={this.fetchChannelConfiguration}>Fetch</button>
            <span>{this.state.fetchStatus}</span>
          </>}
          <label className="configuration-service-view-property">
            <div className="configuration-service-view-property__name">Configuration</div>
            <textarea className={configurationClassName} name="configuration" value={this.state.configuration} onChange={this.onChange} />
          </label>
          <label className="configuration-service-view-property">
            <div className="configuration-service-view-property__name">Version</div>
            <input className={versionClassName} type="text" name="version" value={this.state.version} onChange={this.onChange} />
          </label>
          <button className="configuration-service-view__button" onClick={this.save}>Save</button>
          <button className="configuration-service-view__button" onClick={this.cancel}>Cancel</button>
        </div>
        <div className="configuration-service-view__vertical-bar" />
        <div className="configuration-service-view__section configuration-service-view__section--right">
          <div className="configuration-service-view__title">Using the Configuration Service</div>
          <p className="configuration-service-view__text">The Configuration Service can be used in
            your extension by leveraging specific APIs in the Extension Helper Library.</p>
          <p className="configuration-service-view__text">To access, view, and directly edit your
            data in the Rig, pick Broadcaster, Developer, or Global, and if appropriate, search
            for a specific broadcaster who has installed your extension.  Alternatively, when
            testing your extension in the Extension Views tab, your extension front-end will be
            able to access the object returned by the Configuration Service on extension load.</p>
          <p className="configuration-service-view__text">Having Trouble?  Make sure youâ€™re set to
            use Hosted Configuration
            on <a href={extensionUrl} target="dev-site">your Extension Capabilities page</a>.
            Also you might need to refresh your Extension Manifest in the Rig.  You can do so on
            the Project Tab.  Happy Coding!</p>
          <button className="configuration-service-view__button configuration-service-view__button--first" onClick={this.viewTutorial}>View Tutorial</button>
          <button className="configuration-service-view__button" onClick={this.viewDocumentation}>Go to Documentation</button>
        </div>
      </div>
    );
  }
}
