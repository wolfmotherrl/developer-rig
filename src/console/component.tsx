import * as React from 'react';
import './component.sass';

interface ConsoleEntry {
  frame: string;
  log: string;
}

interface State {
  logHistory: ConsoleEntry[];
}

export class Console extends React.Component<{}, State> {
  private console: React.RefObject<HTMLDivElement>;

  public state: State = {
    logHistory: window.rig.history || [],
  }

  constructor(props: {}) {
    super(props);
    this.console = React.createRef();
  }

  private updateConsole() {
    setTimeout(() => {
      const console = this.console.current;
      if (console) {
        const { offsetHeight, scrollHeight } = console;
        console.scrollTop = Math.max(0, scrollHeight - offsetHeight);
      }
    });
    this.setState({
      logHistory: window.rig.history
    });
  }

  public componentDidMount() {
    window.rig.update = () => {
      this.updateConsole();
    }

    this.updateConsole();
  }

  public render() {
    const logs = this.state.logHistory.map((entry, index) => (
      <div key={index}>{entry.frame} $ {entry.log}</div>
    ));

    return (
      <div className="console" ref={this.console}>{logs}</div>
    );
  }
}
