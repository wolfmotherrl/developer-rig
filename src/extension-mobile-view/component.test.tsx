import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { ExtensionMobileView } from './component';
import { createExtensionForTest } from '../tests/constants/extension';
import { MobileOrientation } from '../constants/mobile';
import { ExtensionMode } from '../constants/extension-coordinator';

const setupShallow = setupShallowTest(ExtensionMobileView, () => ({
  channelId: 'twitch',
  configuration: {},
  id: '0',
  extension: createExtensionForTest(),
  orientation: MobileOrientation.Portrait,
  frameSize: { width: 100, height: 100 },
  position: { x: 0, y: 0 },
  role: ExtensionMode.Viewer,
  isLocal: false,
  bindIframeToParent: jest.fn(),
  installationAbilities: {
    isChatEnabled: true,
  }
}));

describe('<ExtensionMobileView />', () => {
  it('renders correctly', () => {
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it('computes correct styles in portrait mode', () => {
    const { wrapper } = setupShallow();
    const instance = wrapper.instance() as ExtensionMobileView;
    const expectedFrameStyles = {
      'bottom': '0',
      'height': '65px',
      'position': 'absolute',
      'width': '100px'
    };
    const expectedViewStyles = {
      'background': '#322F37',
      'height': '100px',
      'width': '100px',
    };

    expect(instance.computeFrameStyles()).toEqual(expectedFrameStyles);
    expect(instance.computeViewStyles()).toEqual(expectedViewStyles);
  });

  it('computes correct styles in landscape mode', () => {
    const { wrapper } = setupShallow({
      orientation: MobileOrientation.Landscape
    });
    const instance = wrapper.instance() as ExtensionMobileView;

    const expectedFrameStyles = {
      'height': '100px',
      'position': 'absolute',
      'right': '0',
      'width': '28px'
    };

    const expectedViewStyles = {
      'background': '#322F37',
      'height': '100px',
      'width': '100px',
    };

    expect(instance.computeFrameStyles()).toEqual(expectedFrameStyles);
    expect(instance.computeViewStyles()).toEqual(expectedViewStyles);
  })
});
