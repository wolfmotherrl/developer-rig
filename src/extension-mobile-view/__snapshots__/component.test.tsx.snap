// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExtensionMobileView /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionMobileView
    bindIframeToParent={[Function]}
    channelId="twitch"
    configuration={Object {}}
    extension={
        Object {
            "authorName": "test",
            "bitsEnabled": false,
            "clientId": "mockClientId",
            "description": "description",
            "hasChatSupport": false,
            "iconUrl": "iconUrl",
            "iconUrls": Object {
              "square100": "100x100",
            },
            "id": "id",
            "name": "name",
            "requestIdentityLink": false,
            "sku": "sku",
            "state": "Released",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
                "zoom": false,
                "zoomPixels": 100,
              },
              "config": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "liveConfig": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "mobile": Object {
                "viewerUrl": "https://test:8080",
              },
              "panel": Object {
                "canLinkExternalContent": false,
                "height": 300,
                "viewerUrl": "https://test:8080",
              },
              "videoOverlay": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameSize={
        Object {
            "height": 100,
            "width": 100,
          }
    }
    id="0"
    installationAbilities={
        Object {
            "isChatEnabled": true,
          }
    }
    isLocal={false}
    orientation="Portrait"
    position={
        Object {
            "x": 0,
            "y": 0,
          }
    }
    role="viewer"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        style={
                Object {
                        "bottom": "0",
                        "height": "65px",
                        "position": "absolute",
                        "width": "100px",
                      }
        }
>
        <ExtensionFrame
                bindIframeToParent={[Function]}
                channelId="twitch"
                className="view"
                configuration={Object {}}
                extension={
                        Object {
                                "authorName": "test",
                                "bitsEnabled": false,
                                "clientId": "mockClientId",
                                "description": "description",
                                "hasChatSupport": false,
                                "iconUrl": "iconUrl",
                                "iconUrls": Object {
                                  "square100": "100x100",
                                },
                                "id": "id",
                                "name": "name",
                                "requestIdentityLink": false,
                                "sku": "sku",
                                "state": "Released",
                                "summary": "summary",
                                "token": "token",
                                "vendorCode": "vendorCode",
                                "version": "0.1",
                                "views": Object {
                                  "component": Object {
                                    "aspectHeight": 3000,
                                    "aspectWidth": 2500,
                                    "canLinkExternalContent": false,
                                    "viewerUrl": "https://test:8080",
                                    "zoom": false,
                                    "zoomPixels": 100,
                                  },
                                  "config": Object {
                                    "canLinkExternalContent": false,
                                    "viewerUrl": "https://test:8080",
                                  },
                                  "liveConfig": Object {
                                    "canLinkExternalContent": false,
                                    "viewerUrl": "https://test:8080",
                                  },
                                  "mobile": Object {
                                    "viewerUrl": "https://test:8080",
                                  },
                                  "panel": Object {
                                    "canLinkExternalContent": false,
                                    "height": 300,
                                    "viewerUrl": "https://test:8080",
                                  },
                                  "videoOverlay": Object {
                                    "canLinkExternalContent": false,
                                    "viewerUrl": "https://test:8080",
                                  },
                                },
                                "whitelistedConfigUrls": Array [
                                  "foo",
                                ],
                                "whitelistedPanelUrls": Array [
                                  "bar",
                                ],
                              }
                }
                frameId="frameid-0"
                installationAbilities={
                        Object {
                                "isChatEnabled": true,
                              }
                }
                isLocal={false}
                isPopout={false}
                mode="viewer"
                type="mobile"
        />
</div>,
      "className": "view component-view",
      "style": Object {
        "background": "#322F37",
        "height": "100px",
        "width": "100px",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ExtensionFrame
          bindIframeToParent={[Function]}
          channelId="twitch"
          className="view"
          configuration={Object {}}
          extension={
                    Object {
                              "authorName": "test",
                              "bitsEnabled": false,
                              "clientId": "mockClientId",
                              "description": "description",
                              "hasChatSupport": false,
                              "iconUrl": "iconUrl",
                              "iconUrls": Object {
                                "square100": "100x100",
                              },
                              "id": "id",
                              "name": "name",
                              "requestIdentityLink": false,
                              "sku": "sku",
                              "state": "Released",
                              "summary": "summary",
                              "token": "token",
                              "vendorCode": "vendorCode",
                              "version": "0.1",
                              "views": Object {
                                "component": Object {
                                  "aspectHeight": 3000,
                                  "aspectWidth": 2500,
                                  "canLinkExternalContent": false,
                                  "viewerUrl": "https://test:8080",
                                  "zoom": false,
                                  "zoomPixels": 100,
                                },
                                "config": Object {
                                  "canLinkExternalContent": false,
                                  "viewerUrl": "https://test:8080",
                                },
                                "liveConfig": Object {
                                  "canLinkExternalContent": false,
                                  "viewerUrl": "https://test:8080",
                                },
                                "mobile": Object {
                                  "viewerUrl": "https://test:8080",
                                },
                                "panel": Object {
                                  "canLinkExternalContent": false,
                                  "height": 300,
                                  "viewerUrl": "https://test:8080",
                                },
                                "videoOverlay": Object {
                                  "canLinkExternalContent": false,
                                  "viewerUrl": "https://test:8080",
                                },
                              },
                              "whitelistedConfigUrls": Array [
                                "foo",
                              ],
                              "whitelistedPanelUrls": Array [
                                "bar",
                              ],
                            }
          }
          frameId="frameid-0"
          installationAbilities={
                    Object {
                              "isChatEnabled": true,
                            }
          }
          isLocal={false}
          isPopout={false}
          mode="viewer"
          type="mobile"
/>,
        "style": Object {
          "bottom": "0",
          "height": "65px",
          "position": "absolute",
          "width": "100px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bindIframeToParent": [Function],
          "channelId": "twitch",
          "className": "view",
          "configuration": Object {},
          "extension": Object {
            "authorName": "test",
            "bitsEnabled": false,
            "clientId": "mockClientId",
            "description": "description",
            "hasChatSupport": false,
            "iconUrl": "iconUrl",
            "iconUrls": Object {
              "square100": "100x100",
            },
            "id": "id",
            "name": "name",
            "requestIdentityLink": false,
            "sku": "sku",
            "state": "Released",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
                "zoom": false,
                "zoomPixels": 100,
              },
              "config": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "liveConfig": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "mobile": Object {
                "viewerUrl": "https://test:8080",
              },
              "panel": Object {
                "canLinkExternalContent": false,
                "height": 300,
                "viewerUrl": "https://test:8080",
              },
              "videoOverlay": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          },
          "frameId": "frameid-0",
          "installationAbilities": Object {
            "isChatEnabled": true,
          },
          "isLocal": false,
          "isPopout": false,
          "mode": "viewer",
          "type": "mobile",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          style={
                    Object {
                              "bottom": "0",
                              "height": "65px",
                              "position": "absolute",
                              "width": "100px",
                            }
          }
>
          <ExtensionFrame
                    bindIframeToParent={[Function]}
                    channelId="twitch"
                    className="view"
                    configuration={Object {}}
                    extension={
                              Object {
                                        "authorName": "test",
                                        "bitsEnabled": false,
                                        "clientId": "mockClientId",
                                        "description": "description",
                                        "hasChatSupport": false,
                                        "iconUrl": "iconUrl",
                                        "iconUrls": Object {
                                          "square100": "100x100",
                                        },
                                        "id": "id",
                                        "name": "name",
                                        "requestIdentityLink": false,
                                        "sku": "sku",
                                        "state": "Released",
                                        "summary": "summary",
                                        "token": "token",
                                        "vendorCode": "vendorCode",
                                        "version": "0.1",
                                        "views": Object {
                                          "component": Object {
                                            "aspectHeight": 3000,
                                            "aspectWidth": 2500,
                                            "canLinkExternalContent": false,
                                            "viewerUrl": "https://test:8080",
                                            "zoom": false,
                                            "zoomPixels": 100,
                                          },
                                          "config": Object {
                                            "canLinkExternalContent": false,
                                            "viewerUrl": "https://test:8080",
                                          },
                                          "liveConfig": Object {
                                            "canLinkExternalContent": false,
                                            "viewerUrl": "https://test:8080",
                                          },
                                          "mobile": Object {
                                            "viewerUrl": "https://test:8080",
                                          },
                                          "panel": Object {
                                            "canLinkExternalContent": false,
                                            "height": 300,
                                            "viewerUrl": "https://test:8080",
                                          },
                                          "videoOverlay": Object {
                                            "canLinkExternalContent": false,
                                            "viewerUrl": "https://test:8080",
                                          },
                                        },
                                        "whitelistedConfigUrls": Array [
                                          "foo",
                                        ],
                                        "whitelistedPanelUrls": Array [
                                          "bar",
                                        ],
                                      }
                    }
                    frameId="frameid-0"
                    installationAbilities={
                              Object {
                                        "isChatEnabled": true,
                                      }
                    }
                    isLocal={false}
                    isPopout={false}
                    mode="viewer"
                    type="mobile"
          />
</div>,
        "className": "view component-view",
        "style": Object {
          "background": "#322F37",
          "height": "100px",
          "width": "100px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ExtensionFrame
            bindIframeToParent={[Function]}
            channelId="twitch"
            className="view"
            configuration={Object {}}
            extension={
                        Object {
                                    "authorName": "test",
                                    "bitsEnabled": false,
                                    "clientId": "mockClientId",
                                    "description": "description",
                                    "hasChatSupport": false,
                                    "iconUrl": "iconUrl",
                                    "iconUrls": Object {
                                      "square100": "100x100",
                                    },
                                    "id": "id",
                                    "name": "name",
                                    "requestIdentityLink": false,
                                    "sku": "sku",
                                    "state": "Released",
                                    "summary": "summary",
                                    "token": "token",
                                    "vendorCode": "vendorCode",
                                    "version": "0.1",
                                    "views": Object {
                                      "component": Object {
                                        "aspectHeight": 3000,
                                        "aspectWidth": 2500,
                                        "canLinkExternalContent": false,
                                        "viewerUrl": "https://test:8080",
                                        "zoom": false,
                                        "zoomPixels": 100,
                                      },
                                      "config": Object {
                                        "canLinkExternalContent": false,
                                        "viewerUrl": "https://test:8080",
                                      },
                                      "liveConfig": Object {
                                        "canLinkExternalContent": false,
                                        "viewerUrl": "https://test:8080",
                                      },
                                      "mobile": Object {
                                        "viewerUrl": "https://test:8080",
                                      },
                                      "panel": Object {
                                        "canLinkExternalContent": false,
                                        "height": 300,
                                        "viewerUrl": "https://test:8080",
                                      },
                                      "videoOverlay": Object {
                                        "canLinkExternalContent": false,
                                        "viewerUrl": "https://test:8080",
                                      },
                                    },
                                    "whitelistedConfigUrls": Array [
                                      "foo",
                                    ],
                                    "whitelistedPanelUrls": Array [
                                      "bar",
                                    ],
                                  }
            }
            frameId="frameid-0"
            installationAbilities={
                        Object {
                                    "isChatEnabled": true,
                                  }
            }
            isLocal={false}
            isPopout={false}
            mode="viewer"
            type="mobile"
/>,
          "style": Object {
            "bottom": "0",
            "height": "65px",
            "position": "absolute",
            "width": "100px",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bindIframeToParent": [Function],
            "channelId": "twitch",
            "className": "view",
            "configuration": Object {},
            "extension": Object {
              "authorName": "test",
              "bitsEnabled": false,
              "clientId": "mockClientId",
              "description": "description",
              "hasChatSupport": false,
              "iconUrl": "iconUrl",
              "iconUrls": Object {
                "square100": "100x100",
              },
              "id": "id",
              "name": "name",
              "requestIdentityLink": false,
              "sku": "sku",
              "state": "Released",
              "summary": "summary",
              "token": "token",
              "vendorCode": "vendorCode",
              "version": "0.1",
              "views": Object {
                "component": Object {
                  "aspectHeight": 3000,
                  "aspectWidth": 2500,
                  "canLinkExternalContent": false,
                  "viewerUrl": "https://test:8080",
                  "zoom": false,
                  "zoomPixels": 100,
                },
                "config": Object {
                  "canLinkExternalContent": false,
                  "viewerUrl": "https://test:8080",
                },
                "liveConfig": Object {
                  "canLinkExternalContent": false,
                  "viewerUrl": "https://test:8080",
                },
                "mobile": Object {
                  "viewerUrl": "https://test:8080",
                },
                "panel": Object {
                  "canLinkExternalContent": false,
                  "height": 300,
                  "viewerUrl": "https://test:8080",
                },
                "videoOverlay": Object {
                  "canLinkExternalContent": false,
                  "viewerUrl": "https://test:8080",
                },
              },
              "whitelistedConfigUrls": Array [
                "foo",
              ],
              "whitelistedPanelUrls": Array [
                "bar",
              ],
            },
            "frameId": "frameid-0",
            "installationAbilities": Object {
              "isChatEnabled": true,
            },
            "isLocal": false,
            "isPopout": false,
            "mode": "viewer",
            "type": "mobile",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
